doc/source/run_engine_api.rst

```diff
@@ -0,0 +1,56 @@
+.. currentmodule:: bluesky.run_engine
+
+RunEngine API Documentation
+===========================
+
+.. autoclass:: RunEngine
+
+    .. automethod:: __call__
+
+    The RunEngine maintains a callback registry of functions that receive any
+    :doc:`documents` generated by plan execution. These methods add and remove
+    functions from that registry.
+
+    .. automethod:: subscribe
```

Comment:
Actually, sleeping on this, maybe tables _are_ better. But it seems that sphinx doesn't give methods separate pages the way it does for functions. Our choices are (1) automethod style or (2) autosummary with no links, and therefore no access to the details like "Examples" or "See also".